src/common/arrow/arrow_appender.cpp

```diff
@@ -142,9 +142,14 @@ static void InitializeFunctionPointers(ArrowAppendData &append_data, const Logic
 	case LogicalTypeId::INTEGER:
 		InitializeAppenderForType<ArrowScalarData<int32_t>>(append_data);
 		break;
-	case LogicalTypeId::TIME_TZ:
-		InitializeAppenderForType<ArrowScalarData<int64_t, dtime_tz_t, ArrowTimeTzConverter>>(append_data);
+	case LogicalTypeId::TIME_TZ: {
+		if (append_data.options.lossless_conversion) {
+			InitializeAppenderForType<ArrowScalarData<int64_t>>(append_data);
```

Comment:
Interesting that the physical type of `TIME_TZ` is `INT64`, but the `bits` field of `dtime_tz_t` is a [uint64_t](https://github.com/duckdb/duckdb/blob/5bdb091a5d67460da3ca3a89f21b7cdc588d4544/src/include/duckdb/common/types/datetime.hpp#L98). Maybe @hawkfish can shed some light on why?

Reply:
Not sure. `TIME` predates me, so you would have to ask @Mytherin or @hannes . I created the `dtime_tz_t` type when it became clear we needed a new type, and I made it unsigned because it only has positive values. Plus it had internal bit structure, which is always easier with unsigned types. 

Postgres uses a struct with the time and offset values separated, but that seemed like overkill and would require a new physical type. Unfortunately, I got the ordering semantics wrong and you can't sort them as integers but have to use the struct's comparison methods.